messaging/messages.go

```diff
@@ -0,0 +1,72 @@
+package messaging
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/matei-oltean/go-torrent/utils"
+)
+
+// messageType represent the different types of peer messages
+type messageType int
+
+const (
+	choke messageType = iota
+	unchoke
+	interested
+	notInterested
+	have
+	bitfield
+	request
+	piece
+	cancel
+)
+
+// Message represents a message: its type and payload
+type Message struct {
+	Type    messageType
+	Payload []byte
+}
+
+// Read reads a message coming from a connection
+func Read(reader io.Reader) (*Message, error) {
+	// A message is composed as follows:
```

Comment:
as follow*

Reply:
It is always *follows*, see https://dictionary.cambridge.org/dictionary/english/as-follows